src/main/java/com/github/davidmoten/rtree2/geometry/Intersects.java

```diff
@@ -111,6 +111,23 @@ else if (geometry instanceof Rectangle)
         }
     };
 
+    public static final BiPredicate<Geometry, Polygon> geometryIntersectsPolygon = new BiPredicate<Geometry, Polygon>() {
+
+        @Override
+        public boolean test(Geometry geometry, Polygon polygon) {
+            if (geometry instanceof Line)
```

Comment:
missing check on geometry being instanceof Polygon. Also required on other geometryIntersects* methods

Reply:
Is this what you mean (latest commit)?